### PR TITLE
Fix IonValueDeserializer's getNullValue when bean is missing a property

### PR DESCRIPTION
**Pull request by atokuzAmzn**
This PR aims to resolve the issue https://github.com/FasterXML/jackson-dataformats-binary/issues/317

Changes:
- Do not call getEmbeddedObject when current token is END_OBJECT
- Add unit test case exposing the issue

### CODE REVIEW

**Comment by atokuzAmzn** [comment]
@mcliedtke Can you review this?

File: ion/src/test/java/com/fasterxml/jackson/dataformat/ion/ionvalue/IonValueDeserializerTest.java [hunk 1]
```diff
@@ -172,6 +176,38 @@ public void shouldOverrideNullAccessPatternToBeDynamic() {
...
+
+        String input1 = "{required:{}, optional:{}}";
+        MyBean deserializedBean1 = ionObjectMapper.readValue(input1, MyBean.class);
+        assertEquals(ionSystem.newEmptyStruct(), deserializedBean1.required);
+        assertEquals(ionSystem.newEmptyStruct(), deserializedBean1.optional);
+
+        // This deserialization should not fail with missing property
+        String input2 = "{required:{}}";
+        MyBean deserializedBean2 = ionObjectMapper.readValue(input2, MyBean.class);
+        assertEquals(ionSystem.newEmptyStruct(), deserializedBean2.required);
```
Comment (mcliedtke) [inline ion/src/test/java/com/fasterxml/jackson/dataformat/ion/ionvalue/IonValueDeserializerTest.java:208]: For completeness, probably best to assert that optional is a Java-null as well.

**Review by mcliedtke — COMMENTED** [review]
Overall LGTM, one small comment on including an additional test assertion

**Comment by mcliedtke** [comment]
Can you also comment on what branches you believe this change to be added to. Are you targeting for updating 2.13?

**Comment by atokuzAmzn** [comment]
@mcliedtke good catch on missing check. I have added the extra assertion.
Yes we are targeting to update 2.13

**Comment by mcliedtke** [comment]
LGTM, I think these changes can be merged. Given the nature of the change, I think merging to 2.13 is pretty safe. @cowtowncoder do you want to handle the merge itself?

**Comment by atokuzAmzn** [comment]
@cowtowncoder or @mcliedtke 
Can we merge this into 2.13?

**Comment by atokuzAmzn** [comment]
Following up with this merge request @cowtowncoder  @mcliedtke

**Comment by cowtowncoder** [comment]
Ok, I am back from my vacation. @atokuzAmzn If you want this merged against 2.13, please rebase or recreate.
While cherry-picking sometimes works, the usual practice is to start with the oldest branch & merge forward.

**Comment by atokuzAmzn** [comment]
This is my first time working with github. I changed the base of this PR but I am not sure if this is what you wanted @cowtowncoder. If needed I can recreate a new PR based on 2.13

**Comment by cowtowncoder** [comment]
@atokuzAmzn Alas, no: this would merge back all changes from 2.14 -- and not just ones you are making -- so that wouldn't work. I am not 100% sure what'd be the easiest way to change the base for existing PR. Maybe others can suggest something?

But it may be easier to simply create a new PR, if that works?

**Comment by atokuzAmzn** [comment]
I created a new PR based on 2.13 here : https://github.com/FasterXML/jackson-dataformats-binary/pull/322

**Pull request closed without merging**